lib/browser/property-descriptor.ts

```diff
@@ -67,18 +67,39 @@ function canPatchViaPropertyDescriptor() {
   // by default XMLHttpRequest.prototype.onreadystatechange is undefined
   // without adding enumerable and configurable will cause onreadystatechange
   // non-configurable
-  Object.defineProperty(XMLHttpRequest.prototype, 'onreadystatechange', {
-    enumerable: true,
-    configurable: true,
-    get: function() {
-      return true;
-    }
-  });
-  const req = new XMLHttpRequest();
-  const result = !!req.onreadystatechange;
-  // restore original desc
-  Object.defineProperty(XMLHttpRequest.prototype, 'onreadystatechange', xhrDesc || {});
-  return result;
+  // and if XMLHttpRequest.prototype.onreadystatechange is undefined,
+  // we should set a real desc instead a fake one
+  if (xhrDesc) {
```

Comment:
It would probably make sense to just immediately return false if xhrDesc is falsy...

```
if (!xhrDesc) return;
```
this reduces nesting